zebra-network/src/address_book.rs

```diff
@@ -304,6 +304,8 @@ impl AddressBook {
         self.by_addr
             .values()
             .filter(|peer| peer.is_ready_for_connection_attempt())
+            // SECURITY: do not keep trying to reconnect to peers that are likely unreachable.
+            .filter(|peer| !peer.is_probably_unreachable())
```

Comment:
Naming: It's harder to read code that uses double-negatives. And it's easy to miss the initial `!` operator.

We'd need to invert the logic in `is_probably_reachable` as well.

```rust
            .filter(|peer| peer.is_probably_reachable())
```

Reply:
Done in [b533941](https://github.com/ZcashFoundation/zebra/pull/3030/commits/b533941f80fbae403d2e7d3cf6cd7b2c6f231406).